tests/admin/test-class-my-yoast-route.php

```diff
@@ -0,0 +1,299 @@
+	}
+
+	/**
+	 * Tests handling the route.
+	 *
+	 * @covers WPSEO_MyYoast_Route::handle_route
+	 */
+	public function test_handle_route() {
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Route' )
+			->setMethods( array( 'get_action', 'connect' , 'stop_execution' ) )
+			->getMock();
+
+		$instance
+			->expects( $this->once() )
+			->method( 'get_action' )
+			->will( $this->returnValue( 'connect' ) );
+
+		$instance
+			->expects( $this->once() )
+			->method( 'connect' );
+
+		$instance->handle_route();
+	}
+
+	/**
+	 * Tests the is myyoast route check.
+	 *
+	 * @covers WPSEO_MyYoast_Route::is_myyoast_route
+	 */
+	public function test_is_myyoast_route() {
+		$instance = new WPSEO_MyYoast_Route_Double();
+
+		$this->assertTrue( $instance->is_myyoast_route( 'wpseo_myyoast' ) );
+		$this->assertFalse( $instance->is_myyoast_route( 'another_route' ) );
+	}
+
+	/**
+	 * Tests if the given action is a valid action.
+	 *
+	 * @covers WPSEO_MyYoast_Route::is_valid_action
+	 */
+	public function test_is_valid_action() {
+		$instance = new WPSEO_MyYoast_Route_Double();
+
+		$this->assertTrue( $instance->is_valid_action( 'connect' ) );
+		$this->assertFalse( $instance->is_valid_action( 'invalid_action' ) );
+	}
+
+	/**
+	 * Tests connecting with having a saved clientid.
+	 *
+	 * @covers WPSEO_MyYoast_Route::connect
+	 */
+	public function test_connect_with_a_saved_clientid() {
+		$client = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Client' )
+			->setMethods( array( 'get_configuration', 'save_configuration' ) )
+			->getMock();
+
+		$client
+			->expects( $this->once() )
+			->method( 'get_configuration' )
+			->will(
+				$this->returnValue(
+					array(
+						'clientId' => '9740f9cf-608e-4327-8a16-24e3ff6a4c0d',
+						'secret'   => NULL,
+					)
+				)
+			);
+
+		// If the clientId is set no configuration should be saved.
+		$client
+			->expects( $this->never() )
+			->method( 'save_configuration' );
+
+		$instance = $this
+			->getMockBuilder( 'WPSEO_MyYoast_Route_Double' )
+			->setMethods( array( 'redirect', 'get_client' ) )
+			->getMock();
+
+		// Regardless of clientId, redirect should be called.
+		$instance
+			->expects( $this->once() )
+			->method( 'redirect' )
+			->with(
+				'https://my.yoast.com/connect',
+				array(
+					'url'          => WPSEO_Utils::get_home_url(),
+					'client_id'    => '9740f9cf-608e-4327-8a16-24e3ff6a4c0d',
+					'extensions'   => array(),
+					'redirect_url' => admin_url( 'admin.php?page=' . WPSEO_Admin::PAGE_IDENTIFIER ),
+				)
+			);
+
+		$instance
+			->expects( $this->once() )
+			->method( 'get_client' )
+			->will( $this->returnValue( $client ) );
+
+		/**
+		 * @var WPSEO_MyYoast_Route_Double $instance
+		 */
+		$instance->connect();
+	}
+
+	/**
+	 * Tests connecting without having a saved clientid.
+	 *
+	 * @covers WPSEO_MyYoast_Route::connect
+	 */
+	public function test_connect_with_no_saved_clientid() {
```

Comment:
Can be simplified as regardless of whether any info was saved a new client ID should be generated.